module/dkg/client.go

```diff
@@ -0,0 +1,218 @@
+package dkg
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"google.golang.org/grpc"
+
+	"github.com/onflow/cadence"
+
+	"github.com/onflow/flow-core-contracts/lib/go/templates"
+
+	sdk "github.com/onflow/flow-go-sdk"
+	sdkcrypto "github.com/onflow/flow-go-sdk/crypto"
+
+	"github.com/onflow/flow-go/crypto"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/model/messages"
+	"github.com/onflow/flow-go/module"
+)
+
+// Client is a client to the Flow DKG contract. Allows functionality to Broadcast,
+// read a Broadcast and submit the final result of the DKG protocol
+type Client struct {
+	dkgContractAddress string
+	accountAddress     sdk.Address
+	accountKeyIndex    uint
+	flowClient         module.DKGSDKClientWrapper
+	signer             sdkcrypto.Signer
+
+	env templates.Environment // the required contract addresses to be used by the Flow SDK
+}
+
+// NewClient initializes a new client to the Flow DKG contract
+func NewClient(flowClient module.DKGSDKClientWrapper, signer sdkcrypto.Signer, dkgContractAddress, accountAddress string, accountKeyIndex uint) *Client {
+	return &Client{
+		flowClient:         flowClient,
+		dkgContractAddress: dkgContractAddress,
+		signer:             signer,
+		accountKeyIndex:    accountKeyIndex,
+		accountAddress:     sdk.HexToAddress(accountAddress),
+		env:                templates.Environment{DkgAddress: dkgContractAddress},
+	}
+}
+
+// Broadcast broadcasts a message to all other nodes participating in the
+// DKG. The message is broadcast by submitting a transaction to the DKG
+// smart contract. An error is returned if the transaction has failed has
+// failed
+func (c *Client) Broadcast(msg messages.DKGMessage) error {
+
+	ctx := context.Background()
+
+	// get account for given address
+	account, err := c.flowClient.GetAccount(ctx, c.accountAddress, grpc.EmptyCallOption{})
+	if err != nil {
+		return fmt.Errorf("could not get account details: %v", err)
+	}
+
+	// get latest sealed block to execute transaction
+	latestBlock, err := c.flowClient.GetLatestBlock(ctx, true)
+	if err != nil {
+		return fmt.Errorf("could not get latest block from node: %w", err)
+	}
+
+	tx := sdk.NewTransaction().
+		SetScript(templates.GenerateSendDKGWhiteboardMessageScript(c.env)).
+		SetGasLimit(9999).
+		SetReferenceBlockID(latestBlock.ID).
+		SetProposalKey(c.accountAddress, int(c.accountKeyIndex), account.Keys[int(c.accountKeyIndex)].SequenceNumber).
+		SetPayer(c.accountAddress).
+		AddAuthorizer(c.accountAddress)
+
+	err = tx.AddArgument(cadence.NewString(string(msg.Data)))
```

Comment:
I think we should convert this (and binary data we're sending to Cadence in general) to hex. It will be easier to debug -- I'm personally not sure what to expect this string to look like with just the type conversion, then sending to Cadence.